filebeat/input/file.go

```diff
@@ -75,6 +85,29 @@ func (f *FileEvent) ToMapStr() common.MapStr {
 		"count":      1,
 	}
 
+	if f.jsonDecoder != nil {
+		var jsonObj common.MapStr
+		err := json.Unmarshal([]byte(*f.Text), &jsonObj)
+		if err != nil {
+			logp.Err("Error decoding JSON: %v", err)
+			if f.jsonDecoder.OnUnmarshalError == "add_error_key" {
+				event["json_error"] = fmt.Sprintf("Error decoding JSON: %v", err)
```

Comment:
An other option could be to introduce a general "error" field under `beat.error` that we can also use in the future for other errors. This would make it possible to then search for all log events with an error.